packages/flutter/lib/src/rendering/object.dart

```diff
@@ -4583,133 +4328,609 @@ mixin RelayoutWhenSystemFontsChangeMixin on RenderObject {
 
-  /// Disallows this fragment to merge any configuration into its parent's
-  /// [SemanticsNode].
+  /// The cached node created directly by this Object.
   ///
-  /// After calling this, the fragment will only produce children to be added
-  /// to the parent and it will return null for [config].
-  void markAsExplicit();
-
-  /// Consume the fragments of children.
+  /// This cache is filled after the `_SemanticsFragment` provided by this
+  /// object is compiled and forms a semantics node.
   ///
-  /// For each provided fragment it will add that fragment's children to
-  /// this fragment's children (as returned by [compileChildren]) and merge that
-  /// fragment's [config] into this fragment's [config].
+  /// Caching the semantics node ensures the id is consistent in the life time
+  /// of this object.
   ///
-  /// If a provided fragment should not merge anything into [config] call
-  /// [markAsExplicit] before passing the fragment to this method.
-  @override
-  void addAll(Iterable<_InterestingSemanticsFragment> fragments);
-
-  /// Whether this fragment wants to add any semantic information to the parent
-  /// [SemanticsNode].
-  bool get hasConfigForParent => config != null;
+  /// Note, not all `_SemanticsFragment` forms a semantics node. See
+  /// the `_SwitchableSemanticsFragment`.
+  SemanticsNode? cachedSemanticsNode;
 
-  @override
-  List<_InterestingSemanticsFragment> get mergeUpFragments => <_InterestingSemanticsFragment>[this];
+  // Use [semanticsConfiguration] to access.
+  SemanticsConfiguration? _cachedSemanticsConfiguration;
+  SemanticsConfiguration get semanticsConfiguration {
+    if (_cachedSemanticsConfiguration == null) {
+      _cachedSemanticsConfiguration = SemanticsConfiguration();
+      renderObject.describeSemanticsConfiguration(_cachedSemanticsConfiguration!);
+      assert(
+      !_cachedSemanticsConfiguration!.explicitChildNodes || _cachedSemanticsConfiguration!.childConfigurationsDelegate == null,
+      'A SemanticsConfiguration with explicitChildNode set to true cannot have a non-null childConfigsDelegate.',
+      );
+    }
+    return _cachedSemanticsConfiguration!;
+  }
 
-  Set<SemanticsTag>? _tagsForChildren;
+  // Use [getFragment] or [_getSemanticsForParent] to access.
+  _SemanticsFragment? _cachedSemanticFragment;
 
-  /// Tag all children produced by [compileChildren] with `tags`.
+  /// Updates caches stored in this object.
   ///
-  /// `tags` must not be empty.
-  void addTags(Iterable<SemanticsTag> tags) {
-    assert(tags.isNotEmpty);
-    _tagsForChildren ??= <SemanticsTag>{};
-    _tagsForChildren!.addAll(tags);
-  }
+  /// This can only be called if the owning rendering object is a semantics
+  /// boundary. For non boundary rendering objects, they require semantics
+  /// information from both their parent and child rendering objects to update
+  /// its cache, so it can't update by themselves.
```

Comment:
Leave a breadcrumb in this comment explaining how those objects are updated that are not going through this method.